src/authorizer.js

```diff
@@ -90,6 +145,12 @@ class Authorizer {
       },
       context: resolver(identity, token)
     };
+
+    if (this.configuration.usagePlan) {
+      policy.policyDocument.usageIdentifierKey = await this.resolveApiKey(identity.sub);
```

Comment:
I believe we this to be the azp unless the azp doesn't exist, then we'll want to fall back to the sub. That way it is the azp we are rate limiting and not explicit users. If one UI is doing something bad this will protect us from the UI rather than block the user everywhere.

Reply:
Also I don't see `resolveApiKey` function defined, did I miss it?

Reply:
I believe the return result of this function is always the same as the input, which means we have an opportunity to simplify the code. Instead we could write:
```js
policy.policyDocument.usageIdentifierKey = indentify.azp || identity.sub;
await this.ensureApiKey(policy.policyDocument.usageIdentifierKey);
```

Reply:
I disagree, if we would couple it, then we're forcing the user to use rate limiting and we'd make a possibly breaking change